src/main/groovy/com/twcable/grabbit/server/GrabbitContentPullServlet.groovy

```diff
@@ -35,12 +40,94 @@ import static javax.servlet.http.HttpServletResponse.SC_OK
+     *     <li><b>after</b> is ISO-8601 date that is used to stream delta content. It is not required.
+     * </ul>
+     *
+     * {@link GrabbitContentPullServlet} will use the request remote user credentials to authenticate against the server JCR.
+     *
+     * @param request The request to process.
+     * @param response Our response to the request.
+     */
+    @Override
+    void doPost(SlingHttpServletRequest request, SlingHttpServletResponse response) {
+        log.trace "Received a post method"
+
+        try {
+            StringBuilder sb = new StringBuilder();
+            String s;
+            while ((s = request.getReader().readLine()) != null) {
+                sb.append(s);
+            }
+
+            JSONObject jsonObject = null;
+
+            try {
+                jsonObject = new JSONObject(sb.toString());
+            } catch (JSONException e) {
+                log.error("Exception occurred trying to read string into json object", e);
+            }
+
+            log.debug "jsonObject={}", jsonObject
+
+            if (!jsonObject.has("path") || jsonObject.get("path") == null) {
+                log.info "jsonObject doesn't contain path"
+                response.status = SC_BAD_REQUEST
+                response.writer.write("No path provided for content!")
+                return
+            }
+
+            final String path = jsonObject.get("path")
+
+            log.debug "path={}", path
+
+            String afterDateString = "";
+            if (jsonObject.has("after")) {
+                afterDateString = jsonObject.get("after")
+            }
+            log.debug "afterDateString={}", afterDateString
+
+            List<String> excludePathsList = new ArrayList<>();
+            if (jsonObject.has("excludePaths")) {
+                JSONArray excludedPathsJSONArray = jsonObject.getJSONArray("excludePaths");
+                for (int i = 0; i < excludedPathsJSONArray.length(); i++) {
+                    excludePathsList.add((String) excludedPathsJSONArray.get(i));
+                }
+            }
+            log.debug "excludePathsList={}", StringUtils.join(excludePathsList, ",")
+
+            response.contentType = "application/octet-stream"
+            response.status = SC_OK
+
+            //The Login of the user making this request.
+            //This user will be used to connect to JCR.
+            //If the User is null, 'anonymous' will be used to connect to JCR.
+            final serverUsername = request.remoteUser
+
+            log.info "\n\n *** About to call server service with the following to start pushing content to " +
+                    "client***\n\n" +
+                    "path={}\nafter={}\nexcludePathList={}\n\n\n", path,
+                    afterDateString, StringUtils.join(excludePathsList, ",")
+
+            serverService.getContentForRootPath(serverUsername, path, excludePathsList ?: null,
```

Comment:
I agree the need to have a POST here instead of GET due to GET's limitations. However, since this is an internal call (from Grabbit client to Grabbit server), I don't think we need to keep both GET and POST. 

I would thus suggest removing `doGet()` and also updating the `GrabbitContentPullServletSpec` accordingly.

Reply:
Ok will do